llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -111,9 +112,27 @@ void InterBlockScheduling::enterFunction(MachineFunction *MF) {
   // Get ourselves a hazard recognizer
   HR = std::make_unique<AIEHazardRecognizer>(MF->getSubtarget());
 
+  const TargetRegisterInfo *TRI = MF->getSubtarget().getRegisterInfo();
+  LiveRegs MBBLiveness(MF);
+  auto &LiveIns = MBBLiveness.getLiveIns();
+
   // Define our universe of blocks
   for (MachineBasicBlock &MBB : *MF) {
-    Blocks.emplace(&MBB, &MBB);
+    auto Itr = Blocks.emplace(&MBB, &MBB).first;
+    auto &BS = Itr->second;
```

Comment:
Please expand the `auto` when it helps readability. Here I think it could be `BlockState &BS = Itr->second`;